content/articles/pull-requests.md

```diff
@@ -0,0 +1,91 @@
++++
+date = "2019-03-05T14:53:37-07:00"
+title = "Pull Request review process"
++++
+
+Our pull request review process is an ongoing work-in-progress.  This document outlines the timeline, expectations, general points and ongoing issues.  Some of it is particular to how we work (using YouTrack and TeamCity in addition to GitHub) but it may be of use to others.
+
+### The timeline
+
+#### First the author:
+
+1. Check your branch name carefully as this cannot be modified once the PR is opened
+1. Make sure tests are passing on any CI systems used (TeamCity, travis, appveyor)
+1. Create PR (github's [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) feature here is nice if you think it might not quite be ready)
+1. Write a paragraph explaining why the PR is being made.  For bug fixes and small features this might be very small.
+1. Check through the diff to make sure that it seems sensible
+1. Comment directly into the PR on any points of oddness that require explanation additional to code comments
+1. Split the PR into multiple smaller PRs if it is too large
+1. Choose a reviewer and assign them on github
+1. Update YouTrack, if the project uses it
+   - Copy the PR URL as a comment
+   - Set yourself as `Implementer`
+   - Set the reviewer as `Assignee`
+   - Set the state as `Submitted`
+
+#### Then the reviewer
+
+1. Check that you are an appropriate reviewer
+1. Look for:
+   * does the code meet a need or requirement (should be clear between the ticket and the PR paragraph)
+   * formatting, line lengths and lint issues
+   * is the code generally understandable? If not escalate via the PR, slack and finally in person if necessary
+   * does the code look like it would work?
+   * does it pass on CI?
+   * does it work when run locally? (not possible for all components)
+   * how are the tests?  is all new code tested?
+   * are there obvious inefficiencies (if appropriate)
+   * for web apps, check that they render properly
```

Comment:
And does it fulfil the pre-agreed design if existent, otherwise is the UI user-friendly and consistent with our style conventions.